src/gui/runtime/plugin/nvim_gui_shim.vim

```diff
@@ -213,3 +213,17 @@ anoremenu <script> Gui.Treeview.Toggle :call <SID>TreeViewShowToggle()
 function GuiShowContextMenu() range
 	call rpcnotify(0, 'Gui', 'ShowContextMenu')
 endfunction
+
+" Notify Gui of any cursor movement, used by GuiScrollBar
+function GuiCursorMovedUpdateScrollBar()
+	call rpcnotify(0, 'Gui', 'CursorMovedUpdateScrollBar', line('w0'), line('$'), winheight(winnr()))
+endfunction
+
+autocmd CursorMoved,CursorMovedI,VimResized * call GuiCursorMovedUpdateScrollBar()
+
+" Show/Hide the Gui ScrollBar
+function! s:GuiScrollBar(enable) abort
+	call rpcnotify(0, 'Gui', 'SetScrollBarVisible', a:enable)
```

Comment:
:+1: the shim should enable this. If anything because the shim might not be there.

Still not sure if there is a point to check GUI support with something like the existing `ui_has_clipboard()`. Might be worth it for user experience.

Reply:
It shouldn't be possible to enable this without the shim, right?

The scrollbar is off by default, and this function is required to enable. I suppose the user could manually call rpcnotify.